table/server/master/src/main/java/alluxio/master/table/Partition.java

```diff
@@ -59,17 +114,29 @@ public Layout getBaseLayout() {
   /**
    * @return the current layout
    */
-  public synchronized Layout getLayout() {
-    return mTransformedLayout == null ? mBaseLayout : mTransformedLayout;
+  public Layout getLayout() {
+    Transformation transformation = mTransformation.get();
+    return transformation == null ? mBaseLayout : transformation.getLayout();
   }
 
   /**
-   * Sets the transformed layout.
+   * Transform the partition.
    *
+   * @param definition the transformation definition
    * @param layout the transformed layout
    */
-  public synchronized void setTransformedLayout(Layout layout) {
-    mTransformedLayout = layout;
+  public void transform(String definition, Layout layout) {
+    mTransformation.set(new Transformation(definition, layout));
+  }
+
+  /**
+   * @param definition the transformation definition
+   * @return whether the latest transformation of Partition has the same definition
+   */
+  public boolean isTransformed(TransformDefinition definition) {
+    Transformation transformation = mTransformation.get();
+    return transformation != null
+        && transformation.getDefinition().equals(definition.getDefinition());
```

Comment:
should we normalize the definition a little bit for comparing (like case-insensitive, whitespace, etc)?